### ISSUE TITLE
Can't connect to public server

### ISSUE THREAD

**Issue by sallaib** [kolbytn/mindcraft#366]
Using Gemini model, it works in private, but when I want to use it for a public server, it disconnects after few seconds.
Using the launcher with the same user, same user name defined in the gemini.json as well.
Settings checked, ip,port, auth defined

> [ './profiles/gemini.json' ]
MindServer running on port 8080
Client connected
Registering agents: UserName123
Agents registered
Starting agent with profile: ./profiles/gemini.json
Starting agent initialization with profile: ./profiles/gemini.json
Initializing action manager...
Initializing prompter...
Using chat settings: { model: 'gemini-1.5-flash', api: 'google' }
Using embedding settings: { api: 'openai' }
Warning: Failed to initialize embedding model: API key "OPENAI_API_KEY" not found in keys.json or environment variables!
Continuing anyway, using word overlap instead.
Initializing history...
Initializing coder...
Initializing npc controller...
Initializing memory bank...
Initializing self prompter...
Initializing examples...
Examples initialized.
FrenHelp logging into minecraft...
(node:74761) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Client connected
Connected to MindServer
Client disconnected
Agent process exited with code 0 and signal null

**Comment by sallaib** [comment]
It turned out the server is using DDoS protection service so there's no dedicated port, but I have to define one in the settings. any ideas? can I set a port range?

**Comment by CameronAuler** [comment]
@sallaib I am experiencing the same issue. Any updates?

**Comment by sallaib** [comment]
Not really. I might be wrong, but the issue is that (at least for me) the server after the connection, assigns/negotiate a different port (probably IP as well) and it (**mind**craft) can't handle this process.

I'm not sure where to request this feature.

**Comment by yeroc5311** [comment]
dang, even i am stumped! any updates though?

**Comment by sallaib** [comment]
nope, but it would be nice if we could send a feature change request somewhere.

**Comment by uukelele** [comment]
Currently mineflayer doesn't seem to handle the packet that tells it to update it's connection. 

Like in the normal Minecraft client, where you can click on a button (or maybe just join the server and wait a few seconds), your client says "Connecting to server" and you join another server. 


Mineflayer can't (yet) handle this, and either crashes or stays in the same server. 


~~It would be worth opening an issue with mineflayer instead.~~

https://github.com/PrismarineJS/mineflayer/issues/3705

**Comment by Garcke** [comment]
[Disable physics during configuration packet](https://github.com/PrismarineJS/mineflayer/pull/3722) ，May be helpful to you

**Comment by uukelele** [comment]
Closing because we cannot solve this problem. You'd be better off just waiting for https://github.com/PrismarineJS/mineflayer/issues/3705 to be solved.
